### PR TITLE
turn on GoogleLoginPrefsTest

### PR DESCRIPTION
**Pull request by elharo**
@patflynn @nbashirbello

### CODE REVIEW

**Comment by elharo** [comment]
@patflynn @loosebazooka The more I look at this test the more broken it seems.

1. It's really testing the umask on the user's system, nothing in our code at all.
2. It's Unix specific.
3. It assumes the Preferences class is file based, which is not necessarily so. 

I'd like to repolace it with a very different test that doesn't assume any of these things. What do you think?

**Comment by elharo** [comment]
Ping

**Comment by nbashirbello** [comment]
You can replace the test. Like I mentioned earlier the Privacy group just wanted to have a test for login to check that only the user has access to the preferences.

**Comment by elharo** [comment]
Is than an LGTM?

File: google-login-plugin/testSrc/com/google/gct/login/GoogleLoginPrefsTest.java [hunk 1]
```diff
@@ -17,40 +17,16 @@
...
-    BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
-    String line = reader.readLine();
-    inputStream.close();
-    String[] splitOutput = line.split(" ");
-
-    // Check the output
-    //assertTrue(splitOutput.length > 1);
-    //assertEquals(PREFERENCE_FILE_PERMISSION, splitOutput[0]);
+  public void testActiveUser() {
+    GoogleLoginPrefs.saveActiveUser("cheese");
```
Comment (nbashirbello) [inline google-login-plugin/testSrc/com/google/gct/login/GoogleLoginPrefsTest.java]: I think we need  to override the GoogleLoginPrefs.PREFERENCES_PATH so you can have a different preference file during testing.

Comment (nbashirbello) [inline google-login-plugin/testSrc/com/google/gct/login/GoogleLoginPrefsTest.java]: @elharo Why do you think preferences are not stored in a file? I remember I verified this for Unix, Windows and Mac.

Comment (elharo) [inline google-login-plugin/testSrc/com/google/gct/login/GoogleLoginPrefsTest.java]: I can do this. It's non-trival but may be worthwhile. I think it requires a major refactoring of GoogleLoginPrefs so it's non-static and can take an injected Preferences object. Should we consider this?

Also, what, is anything, is our pledge on API compatiblity here? That is, do I have to worry about breaking thrid party clients?

Comment (elharo) [inline google-login-plugin/testSrc/com/google/gct/login/GoogleLoginPrefsTest.java]: http://docs.oracle.com/javase/7/docs/api/java/util/prefs/Preferences.html "A node in a hierarchical collection of preference data. This class allows applications to store and retrieve user and system preference and configuration data. This data is stored persistently in an implementation-dependent backing store. Typical implementations include flat files, OS-specific registries, directory servers and SQL databases. The user of this class needn't be concerned with details of the backing store."

Comment (nbashirbello) [inline google-login-plugin/testSrc/com/google/gct/login/GoogleLoginPrefsTest.java]: Would this work:

PREFERENCES_PATH = ApplicationManager.getApplication().isUnitTestMode() ? "/com/google/gct/login" : "/com/google/gct/login/test";

and we can delete test directory after.

Comment (nbashirbello) [inline google-login-plugin/testSrc/com/google/gct/login/GoogleLoginPrefsTest.java]: This is from back when I researched the preferences, " In Linux, the user’s preferences file is stored in their home directory and in Windows, it uses the registry and in Mac, the user’s preferences file is stored in their home directory."

**Comment by nbashirbello** [comment]
I am fine with just having testActiveUser if we can't test the accessibility of the user preference file.

**Comment by elharo** [comment]
I think 


PREFERENCES_PATH = ApplicationManager.getApplication().isUnitTestMode() ? "/com/google/gct/login" : "/com/google/gct/login/test";

would work. However it's a little icky because it mixes test code into the model code and depeneds on global static state. An alternative is to add a VisibleForTesting setprefernecesPath method, although this is still all static.

**Comment by nbashirbello** [comment]
ok

**Comment by elharo** [comment]
done

**Comment by nbashirbello** [comment]
LGTM

**Pull request merged**